testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/cli/RemoveEJBSubsystemServerSetupTask.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2018, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.test.integration.management.cli;
+
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.test.integration.security.common.CoreUtils;
+import org.jboss.as.test.shared.SnapshotRestoreSetupTask;
+import org.jboss.dmr.ModelNode;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.REMOVE;
+
+/**
+ * Server setup task for test RemoveEJBSubsystemTestCase.
+ * Removes EJB subsystem.
+ *
+ * @author Daniel Cihak
+ */
+public class RemoveEJBSubsystemServerSetupTask extends SnapshotRestoreSetupTask {
+
+    @Override
+    public void doSetup(ManagementClient managementClient, String containerId) throws Exception {
+        // /subsystem=ejb3/service=remote/channel-creation-options=MAX_OUTBOUND_MESSAGES:remove
+        ModelNode removeOp = new ModelNode();
+        removeOp.get(OP).set(REMOVE);
+        removeOp.get(OP_ADDR).add("subsystem", "ejb3");
+        removeOp.get(OP_ADDR).add("service", "remote");
+        removeOp.get(OP_ADDR).add("channel-creation-options", "MAX_OUTBOUND_MESSAGES");
+        CoreUtils.applyUpdate(removeOp, managementClient.getControllerClient());
```

Comment:
Is this class necessary? It seems like this logic could be in the testRemoveEJBSubsystem method itself, which would be clearer in terms of reading the code and would avoid confusing failures in test setup as opposed to the actual test method.  The test could just use @ServerSetup(SnapshotRestoreSetupTask.class)